crypto/src/main/java/org/hyperledger/besu/crypto/NodeKey.java

```diff
@@ -14,16 +14,38 @@
  */
 package org.hyperledger.besu.crypto;
 
-import org.hyperledger.besu.crypto.SECP256K1.PublicKey;
-import org.hyperledger.besu.crypto.SECP256K1.Signature;
-
 import org.apache.tuweni.bytes.Bytes32;
 
-public interface NodeKey {
+public class NodeKey {
+
+  private final SecurityModule crypto;
+
+  public NodeKey(final SecurityModule crypto) {
+    this.crypto = crypto;
+  }
+
+  public SECP256K1.Signature sign(final Bytes32 dataHash) {
+    org.hyperledger.besu.crypto.Signature signature = crypto.sign(dataHash);
+    return SECP256K1.Signature.create(
+        signature.getR(), signature.getS(), signature.getRecoveryId());
+  }
+
+  public PublicKey getPublicKey() {
+    org.hyperledger.besu.crypto.PublicKey pubKey = crypto.getPublicKey();
+    return PublicKey.create(pubKey.getEncoded());
+  }
+
+  public Bytes32 calculateECDHKeyAgreement(final SECP256K1.PublicKey partyKey) {
+    PublicKey pubKey = new org.hyperledger.besu.crypto.PublicKey(partyKey.getEncodedBytes());
 
-  Signature sign(Bytes32 dataHash);
+    return crypto.calculateECDHKeyAgreement(pubKey);
+  }
 
-  PublicKey getPublicKey();
+  public static NodeKey createFrom(final SECP256K1.KeyPair keyPair) {
```

Comment:
This also looks like it is only needed for test code?

Reply:
yeah, its only test ... but simplifies things if its here ... what do you think?

Reply:
A separation of test code from production code would probably be nicer.

Options are either:
- leave as is and annotation with `@VisibleForTesting`  (which has been done elsewhere)
- move into a new NodeKeyTestUtil class that lives among the tests